docs/GettingStarted.md

```diff
@@ -0,0 +1,139 @@
+# Getting Started
+
+CYNQ is as straight-forward as PYNQ. The philosophy behind CYNQ is to be as simple as PYNQ.
+
+1) The first step to integrate CYNQ is to include the header:
+
+```c++
+#include <cynq/cynq.hpp>
+```
+
+2) Create a IHardware object that will generate the proper drivers for handling the accelerator, data mover and memory. It uses the `IHardware::Create(impl, bitstream, xclbin)` factory. For instance:
+
+```c++
+auto impl = cynq::HardwareArchitecture::UltraScale;
+auto platform = 
+    cynq::IHardware::Create(impl, bitstream, xclbin);
+```
+
+where:
+
+* `impl` is the HardwareArchitecture. At the moment, only `UltraScale` is supported for UltraScale MPSoC.
+* `bitstream` is the path to the .bit file (bitstream)
+* `xclbin` is the path to the .xclbin file. You can use the one in `third-party/resources/default.xclbin`.
+
+3) Create the DMA instances to move the data. This is intended for designs that uses AXI4-Stream.
+
+```c++
+constexpr int kDmaAddress = 0xa0010000;
+auto dma = platform->GetDataMover(kDmaAddress);
+```
+
+where `kDmaAddress` is the address of the DMA instance we want to control. This is given by the design.
+
+4) Create the IP core instances to interact with them.
+
+```c++
+constexpr int kAccelAddress = 0xa0000000;
+auto accel = platform->GetAccelerator(kAccelAddress);
+```
+
+where `kAccelAddress` is the AXI4-Lite control port of the IP core. As a requirement:
+
+> The IP Cores must have an AXI4-Lite port as a control port
+
+5) Get buffers to exchange data. These buffers are usually dual memory: they are mapped into host and device regions (physically contiguous).
```

Comment:
make examples of how these memory addresses look like, maybe making a diagram for this in the future can help illustrate this.

Reply:
Addressing it in other PR